airflow/io/__init__.py

```diff
@@ -32,21 +32,25 @@
 if TYPE_CHECKING:
     from fsspec import AbstractFileSystem
 
+    from airflow.io.typedef import Properties
```

Comment:
Why does this need a dedicated type alias? Do we plan to make it stricter with say TypedDict? It feels to me it’s better to use `dict[str, str]` instead; slightly repetitive but cognatively easier for readers.